From 06408aee9d19728a2dbb02ef09171358e46fda10 Mon Sep 17 00:00:00 2001
From: "junliang.zhou" <junliang.zhou@amlogic.com>
Date: Tue, 23 Sep 2014 10:03:13 +0800
Subject: [PATCH 4900/5965] PD #97273:miracast:clear video pts up flag

Squashed commit of the following:

commit 181d0cd3ca9ddd20ca9ec1c03b921b94766e67c4
Author: junliang.zhou <junliang.zhou@amlogic.com>
Date:   Fri Sep 19 13:34:21 2014 +0800

    PD #97723:dont break when amlvideo task is not ready

    Change-Id: I0a4862923b59a8a24ca74572bca7e37030562120

commit 54732609fe952dc892e27e558fd09839d43c8753
Author: junliang.zhou <junliang.zhou@amlogic.com>
Date:   Fri Sep 19 11:51:10 2014 +0800

    PD #97723:dont break when amlvideo task is not ready

    Change-Id: I0c6779089a324b9420c91096ed2982874d448974

commit 9e1564c6a2b233b97237f3297125e9e05ff481a6
Author: junliang.zhou <junliang.zhou@amlogic.com>
Date:   Fri Sep 12 14:58:28 2014 +0800

    PD #97273:clear video pts up flag,move function declare to .h

    Change-Id: Iddd2f0f39649c81b7d527643984924655f422d14

commit 473a58285d9413995dc15bcf06f38b14739b1098
Author: junliang.zhou <junliang.zhou@amlogic.com>
Date:   Fri Sep 12 14:07:53 2014 +0800

    PD #97273:clear video pts up flag

    Change-Id: I6162651eba6f1e12e43039be539a81a2223d80ad

Change-Id: I44c8012b95643777dfcb97e1927327e9f63a0e7e
---
 drivers/amlogic/amports/amports_priv.h | 1 +
 drivers/amlogic/amports/amstream.c     | 2 ++
 drivers/amlogic/amports/video.c        | 7 +++++++
 drivers/amlogic/video_dev/amlvideo2.c  | 4 +++-
 4 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/amports_priv.h b/drivers/amlogic/amports/amports_priv.h
index 565a464dc6a1..d81fdd0ae53e 100755
--- a/drivers/amlogic/amports/amports_priv.h
+++ b/drivers/amlogic/amports/amports_priv.h
@@ -12,6 +12,7 @@ int ext_get_cur_video_frame(vframe_t **vf,int *canvas_index);
 int ext_put_video_frame(vframe_t *vf);
 int ext_register_end_frame_callback(struct amvideocap_req *req);
 int request_video_firmware(const char * file_name,char *buf,int size);
+void set_vsync_pts_inc_mode(int inc);
 
 #endif
 
diff --git a/drivers/amlogic/amports/amstream.c b/drivers/amlogic/amports/amstream.c
index 740775d367b7..9b073ac378cc 100755
--- a/drivers/amlogic/amports/amstream.c
+++ b/drivers/amlogic/amports/amstream.c
@@ -81,6 +81,7 @@ u32 amstream_port_num;
 u32 amstream_buf_num;
 
 extern void set_real_audio_info(void *arg);
+
 //#define DATA_DEBUG
 static int use_bufferlevelx10000=10000;
 static int reset_canuse_buferlevel(int level);
@@ -707,6 +708,7 @@ static  int amstream_port_init(stream_port_t *port)
 #endif
 
     tsync_audio_break(0); // clear audio break
+	set_vsync_pts_inc_mode(0); // clear video inc
 
     port->flag |= PORT_FLAG_INITED;
     return 0;
diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index cd61275abfbc..9aa478e562e3 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -2172,6 +2172,13 @@ int get_vsync_pts_inc_mode(void)
 }
 EXPORT_SYMBOL(get_vsync_pts_inc_mode);
 
+void set_vsync_pts_inc_mode(int inc)
+{
+    vsync_pts_inc_upint = inc;
+}
+EXPORT_SYMBOL(set_vsync_pts_inc_mode);
+
+
 #ifdef CONFIG_VSYNC_RDMA
 void vsync_rdma_process(void)
 {
diff --git a/drivers/amlogic/video_dev/amlvideo2.c b/drivers/amlogic/video_dev/amlvideo2.c
index c296dc76767a..5b01add87ed8 100755
--- a/drivers/amlogic/video_dev/amlvideo2.c
+++ b/drivers/amlogic/video_dev/amlvideo2.c
@@ -2436,6 +2436,8 @@ static int amlvideo2_start_thread(struct amlvideo2_fh *fh)
 #endif
 	dprintk(node->vid_dev, 1, "%s\n", __func__);
 
+    dma_q->task_running = 1;
+
 #ifdef MUTLI_NODE
 	dma_q->kthread = kthread_run(amlvideo2_thread, fh, (node->vid==0)?"amlvideo2-0":"amlvideo2-1");
 #else
@@ -2448,7 +2450,7 @@ static int amlvideo2_start_thread(struct amlvideo2_fh *fh)
 	}
 	/* Wakes thread */
 	//wake_up_interruptible(&dma_q->wq);
-	dma_q->task_running = 1;
+
 	dprintk(node->vid_dev, 1, "returning from %s\n", __func__);
 	return 0;
 }
-- 
2.19.0

